src/dag-node.js

```diff
@@ -82,6 +82,9 @@ module.exports = class DAGNode {
 
   // addNodeLink - adds a DAGLink to this node that points to node by a name
   addNodeLink (name, node) {
+    if (!name) {
```

Comment:
Can we fail instead? If the user is passing in `null` or `undefined` or an empty array, there's probably a deeper misunderstanding -- it's unlikely they're passing in such things because they intentionally want the name to be the empty string.